### PR TITLE
Add GitHub script for releases, and publishing users documentation

### PR DESCRIPTION
**Pull request by krashish8**
Changes proposed in this pull request:
- Adds script for automating releases and publishing documentation.
- Adds script for publishing users documentation by manual workflow trigger.

@pgRouting/admins

### CODE REVIEW

**Comment by krashish8** [comment]
**1. Script for automating releases (On pushing the tag)**
  - Users documentation is published
  - Developers documentation is published (only when tag is created for `main` branch)
  - A draft release is created with a default body and the necessary files are uploaded to the release.

Tested on my fork:
- Commits on `gh-pages` branch: https://github.com/krashish8/vrprouting/commits/gh-pages
- Published Release: https://github.com/krashish8/vrprouting/releases

**Comment by krashish8** [comment]
**2. Script for publishing users documentation (Can be manually triggered from GitHub actions)**
- For publishing users documentation for any branch.
- Useful particularly for `develop` branch, for which tags are not created.
- Workflow: https://github.com/krashish8/vrprouting/runs/1865010894?check_suite_focus=true
  (In my fork, `gh-pages` was already updated, so no extra commit was created)

E.g: In my fork, this is how the workflow can be manually triggered on any branch from [Github Actions](https://github.com/krashish8/vrprouting/actions?query=workflow%3A%22Publish+Users+Documentation%22):
image

**Comment by krashish8** [comment]
@cvvergara Please review!

**Review by cvvergara — APPROVED** [review]

File: doc/conf.py.in [hunk 1]
```diff
@@ -315,6 +315,7 @@ pdf_style_path = ['.', '_styles']
 linkcheck_ignore = [
     # might not exist yet (we are generating it!)
     r"https://vrp.pgrouting.org",
+    "https://gis.stackexchange.com/feeds/tag?tagnames=vrprouting&sort=newest",
```
Comment (cvvergara) [inline doc/conf.py.in:318]: maybe with:
```
r"https://gis.stackexchange.com",
```
it will work

**Review by cvvergara — CHANGES_REQUESTED** [review]

**Review by cvvergara — APPROVED** [review]

**Pull request merged**